src/oemof/solph/flows/_investment_flow_block.py

```diff
@@ -945,6 +1048,89 @@ def _objective_expression(self):
 
         return self.costs
 
+    def _evaluate_remaining_value_difference(
+        self,
+        m,
+        p,
+        i,
+        o,
+        end_year_of_optimization,
+        lifetime,
+        interest,
+        nonconvex=False,
+    ):
+        """Evaluate and return the remaining value difference of an investment
+
+        The remaining value difference in the net present values if the asset
+        was to be liquidated at the end of the optimization horizon and the
+        net present value using the original investment expenses.
+
+        Parameters
+        ----------
+        m : oemof.solph.models.Model
+            Optimization model
+
+        p : int
+            Period in which investment occurs
+
+        i : any instance of oemof.solph.components
+            start node of flow
+
+        o : any instance of oemof.solph.components
+            end node of flow
+
+        end_year_of_optimization : int
+            Last year of the optimization horizon
+
+        lifetime : int
+            lifetime of investment considered
+
+        interest : float
+            Demanded interest rate for investment
+
+        nonconvex : bool
+            Indicating whether considered flow is nonconvex.
+        """
+        if m.es.use_remaining_value:
+            if end_year_of_optimization - m.es.periods_years[p] < lifetime:
+                remaining_lifetime = lifetime - (
+                    end_year_of_optimization - m.es.periods_years[p]
+                )
+                remaining_annuity = economics.annuity(
+                    capex=m.flows[i, o].investment.ep_costs[-1],
+                    n=remaining_lifetime,
+                    wacc=interest,
+                )
+                original_annuity = economics.annuity(
+                    capex=m.flows[i, o].investment.ep_costs[p],
+                    n=remaining_lifetime,
+                    wacc=interest,
+                )
+                present_value_factor_remaining = 1 / economics.annuity(
+                    capex=1, n=remaining_lifetime, wacc=interest
+                )
+                if nonconvex:
+                    return (
+                        self.invest[i, o, p]
+                        * (remaining_annuity - original_annuity)
+                        * present_value_factor_remaining
+                        + self.invest_status[i, o, p]
+                        * (
+                            m.flows[i, o].investment.offset[-1]
+                            - m.flows[i, o].investment.offset[p]
+                        )
+                    ) * (1 + m.discount_rate) ** (-end_year_of_optimization)
+                else:
+                    return (
+                        self.invest[i, o, p]
+                        * (remaining_annuity - original_annuity)
+                        * present_value_factor_remaining
+                    ) * (1 + m.discount_rate) ** (-end_year_of_optimization)
```

Comment:
I think, code duplication can be avoided here.:
```suggestion
                convex_investment_costs = (
                    self.invest[i, o, p]
                    * (remaining_annuity - original_annuity)
                    * present_value_factor_remaining
                ) * (1 + m.discount_rate) ** (-end_year_of_optimization)
                if nonconvex:
                    return (
                        convex_investment_cost
                        + self.invest_status[i, o, p]
                        * (
                            m.flows[i, o].investment.offset[-1]
                            - m.flows[i, o].investment.offset[p]
                        )
                    )
                else:
                    return convex_investment_costs
```

Reply:
Yes, your suggestion lacks the discounting for the offset, though (for the case you want to provide costs in nominal terms). But thanks a lot, I'll integrate it.